TidepoolKit/Model/Datum/BolusDatum/TBolusDatum.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  TBolusDatum.swift
+//  TidepoolKit
+//
+//  Created by Larry Kenyon on 8/30/19.
+//  Copyright © 2019 Tidepool Project. All rights reserved.
+//
+
+import Foundation
+
+public class TBolusDatum: TDatum {
+    public typealias InsulinFormulation = TInsulinDatum.Formulation
+
+    public enum SubType: String, Codable {
+        case combination = "dual/square"
+        case extended = "square"
+        case normal
+    }
+
+    public let subType: SubType
+    public var insulinFormulation: InsulinFormulation?
+
+    public init(_ subType: SubType, time: Date, insulinFormulation: InsulinFormulation? = nil) {
+        self.subType = subType
+        self.insulinFormulation = insulinFormulation
```

Comment:
[nit] will the `insulinFormulation` ever be set? Seems like the subclasses don't include this information?

Reply:
The subclasses can directly access this property after initialization. It is not a very commonly used property so I didn't add it to the subclass initializers (but I added it here for a complete initializer).